src/main/java/data/workouts/WorkoutList.java

```diff
@@ -207,39 +207,53 @@ private boolean checkIndexIsWithinRange(int index) {
      * new number of repetitions.
      *
      * @param userArgument The argument entered by user, which includes index of workout to update
      *                     and new number of repetitions.
      * @return updatedWorkout The workout object which has been updated.
      * @throws ArrayIndexOutOfBoundsException For operations which involves index checking.
-     * @throws NumberFormatException If workout index and number of reps could not be parsed into an integer.
-     * @throws WorkoutOutOfRangeException If workout index is out of total number of workout in workoutsList.
-     * @throws InvalidWorkoutException If the value of repetition is not valid.
+     * @throws NumberFormatException          If workout index and number of reps could not be parsed into an integer.
+     * @throws WorkoutOutOfRangeException     If workout index is out of total number of workout in workoutsList.
+     * @throws InvalidWorkoutException        If the value of repetition is not valid.
      */
     public Workout updateWorkout(String userArgument) throws ArrayIndexOutOfBoundsException,
             NumberFormatException, WorkoutOutOfRangeException, InvalidWorkoutException {
+        logger.entering(getClass().getName(), "updateWorkout");
         String[] updateDetails = userArgument.split(" ", 2);
         String indexToUpdateString = updateDetails[0].trim();
         String newNumberOfRepsString = updateDetails[1].trim();
         int indexToUpdate = Integer.parseInt(indexToUpdateString);
-        int newNumberOfReps = Integer.parseInt(newNumberOfRepsString);
+        int newRepsValue = Integer.parseInt(newNumberOfRepsString);
 
         boolean isIndexToUpdateValid = checkIndexIsWithinRange(indexToUpdate);
-        boolean isNewNumberOfRepsValid = checkIfRepsValueIsValid(newNumberOfReps);
+        boolean isNewRepsValueValid = checkIfRepsValueIsValid(newRepsValue);
         String className = this.getClass().getSimpleName();
 
         if (!isIndexToUpdateValid) {
+            logger.log(Level.WARNING, "Workout index to update is out of range!");
             throw new WorkoutOutOfRangeException(className, WorkoutOutOfRangeException.INDEX_VALUE_OUT_OF_RANGE);
         }
-        if (!isNewNumberOfRepsValid) {
+        if (!isNewRepsValueValid) {
+            logger.log(Level.WARNING, "Repetition value is invalid.");
             throw new InvalidWorkoutException(className, InvalidWorkoutException.INVALID_REPS_VALUE_ERROR_MSG);
         }
 
         Workout updatedWorkout = workoutsList.get(indexToUpdate - 1);
-        updatedWorkout.setRepetitions(newNumberOfReps);
+        String exerciseName = updatedWorkout.getExerciseName();
+        boolean isWorkoutExist = checkForExistingWorkout(exerciseName, newRepsValue);
+
+        if (isWorkoutExist) {
+            logger.log(Level.WARNING, "\"" + exerciseName + " (" + newRepsValue + ")\" " +
+                "has already existed in the list. Cannot be updated." );
```

Comment:
(Reported by Gradle's checkstyle module) The closing parenthesis on Line 250 has an extra space before it.